core/app/models/spree/promotion_code.rb

```diff
@@ -0,0 +1,26 @@
+class Spree::PromotionCode < ActiveRecord::Base
+  belongs_to :promotion
+
+  validates :promotion, :value, presence: true
+  validates :usage_limit, numericality: { greater_than: 0, allow_nil: true }
+
+  before_save :normalize_value
+
+  def expired?
+    !!(starts_at && Time.now < starts_at || expires_at && Time.now > expires_at)
+  end
+
+  def eligible?(promotable)
+    !expired? && !usage_limit_exceeded?(promotable)
+  end
+
+  def usage_limit_exceeded?(promotable)
```

Comment:
Unused method argument - `promotable`. If it's necessary, use `_` or `_promotable` as an argument name to indicate that it won't be used. You can also write as `usage_limit_exceeded?(*)` if you want the method to accept any arguments but don't care about them.